src/elements/element.line.js

```diff
@@ -25,84 +23,135 @@ defaults._set('global', {
 	}
 });
 
+function startAtGap(points, spanGaps) {
+	let closePath = true;
+	let previous = points.length && points[0]._view;
+	let index, view, cloned;
+	for (index = 1; index < points.length; ++index) {
+		// If the line has an open path, shift the point array
+		view = points[index]._view;
+		if (!view.skip && previous.skip) {
+			points = points.slice(index).concat(points.slice(0, index));
+			closePath = spanGaps;
+			cloned = true;
+			break;
+		}
+		previous = view;
+	}
+	// If the line has a close path, add the first point again
+	if (closePath) {
+		if (!cloned) {
+			points = points.slice();
+		}
+		points.push(points[0]);
+	}
+	points.closePath = closePath;
+	return points;
+}
+
+function setStyle(ctx, vm) {
+	ctx.lineCap = vm.borderCapStyle;
+	// IE 9 and 10 do not support line dash
+	if (ctx.setLineDash) {
+		ctx.setLineDash(vm.borderDash);
+	}
+	ctx.lineDashOffset = vm.borderDashOffset;
+	ctx.lineJoin = vm.borderJoinStyle;
+	ctx.lineWidth = vm.borderWidth;
+	ctx.strokeStyle = vm.borderColor;
+}
+
+function normalPath(ctx, points, spanGaps) {
+	let move = true;
+	let index, currentVM, previousVM;
+
+	for (index = 0; index < points.length; ++index) {
+		currentVM = points[index]._view;
+
+		if (currentVM.skip) {
+			move = move || !spanGaps;
+			continue;
+		}
+
+		if (move) {
+			ctx.moveTo(currentVM.x, currentVM.y);
+			move = false;
+		} else {
+			helpers.canvas.lineTo(ctx, previousVM, currentVM);
+		}
+		previousVM = currentVM;
+	}
+}
+
+function fastPath(ctx, points, spanGaps) {
+	let move = true;
+	let index, vm, truncX, y, prevX, minY, maxY, lastY;
+
+	for (index = 0; index < points.length; ++index) {
+		vm = points[index]._view;
+
+		if (vm.skip) {
+			move = move || !spanGaps;
+			continue;
+		}
+		truncX = vm.x | 0;
+		y = vm.y;
+
+		if (move) {
+			ctx.moveTo(vm.x, y);
+			minY = maxY = y;
+			move = false;
+		} else if (truncX === prevX) {
+			minY = Math.min(y, minY);
+			maxY = Math.max(y, maxY);
+		} else {
+			if (minY !== maxY) {
+				ctx.lineTo(prevX, maxY);
+				ctx.lineTo(prevX, minY);
+				ctx.moveTo(prevX, lastY);
```

Comment:
There are some minor optimizations that can be done here if for example `minY === lastY` there's no need to move but not sure how much they change things

Reply:
Probably not worth the added complexity.